### PR TITLE
makes golangci-lint more strict

### PR DESCRIPTION
**Pull request by torcolvin**
Add unconvert / unused to stricter linting. Skip code in `rest/debug.go` which is largely unused.

### CODE REVIEW

**Comment by bbrks** [comment]
What level of analysis has been done to determine whether the things being removed are unintentionally unused? (just so I know how much digging to do as part of the review)

**Comment by torcolvin** [comment]
> What level of analysis has been done to determine whether the things being removed are unintentionally unused? (just so I know how much digging to do as part of the review)

I didn't do a huge amount of analysis about unused code, except around new code. If it was unused, I assumed we had accidentally written around it, like I did with [0069df9](https://github.com/couchbase/sync_gateway/pull/6507/commits/0069df9275060b54ae16c973e221c0a6c8e73d27)

File: base/dcp_dest.go [hunk 1]
```diff
@@ -251,10 +244,6 @@ func partitionToVbNo(ctx context.Context, partition string) uint16 {
 	return uint16(vbNo)
 }
 
-func vbNoToPartition(vbNo uint16) string {
-	return vbucketIdStrings[vbNo]
```
Comment (bbrks) [inline base/dcp_dest.go:256]: This array is initialized with 1024 vbucket IDs but also no longer read once this function has gone.
We could do more and rip out the declaration and init whilst we're here.

File: base/dcp_dest.go [hunk 2]
```diff
@@ -29,12 +29,6 @@ import (
 // (Atoi is faster than map lookup when going in the other direction)
 var vbucketIdStrings [1024]string
 
-type cbgtFeedType int8
-
-const (
-	cbgtFeedType_gocb cbgtFeedType = iota
-)
-
 type destFeedType int8
```
Comment (bbrks) [inline base/dcp_dest.go]: This also seems unused?

File: db/event.go [hunk 3]
```diff
@@ -108,7 +108,6 @@ const (
 // A compiled JavaScript event function.
 type jsEventTask struct {
 	sgbucket.JSRunner
-	responseType ResponseType
```
Comment (bbrks) [inline db/event.go:111]: `type ResponseType` and the const values could be dropped as well if this goes

**Review by bbrks — DISMISSED** [review]
Everything looks good and intentionally removed - couple of places where we'll get new unused code as soon as we merge so might as well go rip out the other bits.

File: base/dcp_dest.go (hunk 1 above)
Comment (torcolvin) [reply base/dcp_dest.go:256]: Ripped this out with the test.

**Review by bbrks — APPROVED** [review]

**Pull request merged**